include/internal/common.h

```diff
@@ -93,87 +93,142 @@ __owur static inline int ossl_assert_int(int expr, const char *exprstr,
+/*
+ * Loop rolling/unrolling, use by some cipher implementations.
+ * Note that all of these evaluate their parameters multiple
+ * times. We could remove the parens around some of them,
+ * like "(c)" etc., but don't. Many of the macros use the comma
+ * to get a "sequence point" to insure that the right bytes
+ * are shifted with the right amounts.
+ */
+
+# define n2s(c, s) \
+    ((s = (((unsigned int)((c)[0])) << 8) | \
+          (((unsigned int)((c)[1]))     )), \
+          (c) += 2)
+
+# define s2n(s, c) \
+    (((c)[0] = (unsigned char)(((s) >> 8) & 0xff), \
+      (c)[1] = (unsigned char)(((s)     ) & 0xff)), \
+      (c) += 2)
+
+# define n2l3(c, l) \
+    ((l = (((unsigned long)((c)[0])) <<16) | \
+          (((unsigned long)((c)[1])) << 8) | \
+          (((unsigned long)((c)[2]))    )), \
+      (c) += 3)
+
+# define l2n3(l, c) \
+    (((c)[0] = (unsigned char)(((l) >> 16) & 0xff), \
+      (c)[1] = (unsigned char)(((l) >>  8) & 0xff), \
+      (c)[2] = (unsigned char)(((l)      ) & 0xff)), \
+      (c) += 3)
+
+# define c2l(c, l) \
+    (l  =  ((unsigned long)(*((c)++))       ), \
+     l |= (((unsigned long)(*((c)++))) <<  8), \
+     l |= (((unsigned long)(*((c)++))) << 16), \
+     l |= (((unsigned long)(*((c)++))) << 24))
+
+# define l2c(l, c) \
+    (*((c)++) = (unsigned char)(((l)      ) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >>  8) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 16) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 24) & 0xff))
+
+# define n2l(c, l) \
+    (l  = ((unsigned long)(*((c)++))) << 24, \
+     l |= ((unsigned long)(*((c)++))) << 16, \
+     l |= ((unsigned long)(*((c)++))) <<  8, \
+     l |= ((unsigned long)(*((c)++)))       )
+
+# define l2n(l, c) \
+    (*((c)++) = (unsigned char)(((l) >> 24) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 16) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >>  8) & 0xff), \
+     *((c)++) = (unsigned char)(((l)      ) & 0xff))
+
+# define n2l8(c, l) \
+    (l  = ((uint64_t)(*((c)++))) << 56, \
+     l |= ((uint64_t)(*((c)++))) << 48, \
+     l |= ((uint64_t)(*((c)++))) << 40, \
+     l |= ((uint64_t)(*((c)++))) << 32, \
+     l |= ((uint64_t)(*((c)++))) << 24, \
+     l |= ((uint64_t)(*((c)++))) << 16, \
+     l |= ((uint64_t)(*((c)++))) <<  8, \
+     l |= ((uint64_t)(*((c)++)))      )
+
+# define l2n8(l, c) \
+    (*((c)++) = (unsigned char)(((l) >> 56) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 48) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 40) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 32) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 24) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >> 16) & 0xff), \
+     *((c)++) = (unsigned char)(((l) >>  8) & 0xff), \
+     *((c)++) = (unsigned char)(((l)      ) & 0xff) )
+
+# define c2ln(c, l1, l2, n) \
+    { \
+        c += n; \
+        l1 = l2 = 0; \
+        switch (n) { \
+        case 8: l2  = ((unsigned long)(*(--(c)))) << 24; \
```

Comment:
Apparently the fallthrough comments need to stay here and in all the other switches.

Reply:
oops.  I added rc2,rc5 to my builds.